jwst/extract_2d/grisms.py

```diff
@@ -445,20 +446,22 @@ def extract_grism_objects(input_model,
                                                  xcenter_model &
                                                  ycenter_model &
                                                  order_model) | tr
-
-                ext_data = input_model.data[ymin: ymax + 1, xmin: xmax + 1].copy()
-                ext_err = input_model.err[ymin: ymax + 1, xmin: xmax + 1].copy()
-                ext_dq = input_model.dq[ymin: ymax + 1, xmin: xmax + 1].copy()
+                from gwcs.utils import _toindex
+                y_slice = slice(_toindex(ymin), _toindex(ymax))
```

Comment:
In retrospect, I think this is a mistake and causes the actual number of extracted rows/cols to be short by 1. For example, if you have ymin=941.8 and ymax=951.8, the `_toindex` function converts those limits to 942 and 952. Those are in fact the *inclusive* limits of what we want to extract, i.e. a total of 11 rows, 942 through 952, inclusive. But when those are used directly as python slice indexes, you of course end up extracting only 942 through 951, a total of 10 rows. That's why the old lines of code explicitly added 1 to the ymax/xmax values before using them in the python slicing operation. This is leading to extraction boxes being undersized by 1 row/col.

Reply:
I believe it's necessary to again add 1 to `_toindex(ymax)` and `_toindex(xmax)` when constructing `y_slice` and `x_slice`.